src/tls.rs

```diff
@@ -268,6 +268,48 @@ impl fmt::Debug for Identity {
     }
 }
 
+/// A TLS protocol version.
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[non_exhaustive]
+pub enum TlsVersion {
```

Comment:
I'd consider a few minor changes here:

- Make the name `Version`, so it's `reqwest::tls::Version`. I think that reads better...
- Make this an opaque struct, kind of like `http::Version` is. Then each variant is an associated constant instead. For naming, perhaps drop the `v`, so it's just `TLS_1_0`, etc?
- Let's include the docs for each variant.
- Drop SSLv3, because just no XD

Reply:
> Make the name `Version`, so it's `reqwest::tls::Version`. I think that reads better...

It does, but `tls` is a private module so I think it'd collide with `reqwest::Version`? At least if it's re-exported like `Certificate` and friends.

> Drop SSLv3, because just no XD

It could be useful for testing. I've included it in ducaale/xh#168: if `xh --ssl ssl3 https://yourserver` works then you know something is wrong.

Admittedly my own system's OpenSSL doesn't even have it enabled 😛

Reply:
Oh right. Hm, I think originally I tried to make everything flat, but then as options grew, started making specific modules again (`reqwest::redirect::Policy` vs `reqwest::RedirectPolicy`). I'd be fine with making `tls` public...

I still lean towards nuking SSLv3. It's just too bad. Even curl returns a `CURLE_BAD_FUNCTION_ARGUMENT` if you pass SSLv2 or v3.

Reply:
I've made `tls` public but kept the re-exports to not immediately break backward compatibility.